lib/lm3s/usart.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2013 Gareth McMullin <gareth@blacksphere.co.nz>
+ * Copyright (C) 2015 Daniele Lacamera <root at danielinux.net>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <libopencm3/lm3s/usart.h>
+
+void usart_send(uint32_t usart, uint16_t data)
+{
+	USART_DR(usart) = data;
+}
+
+uint16_t usart_recv(uint32_t usart)
+{
+	return USART_DR(usart) & 0xff;
```

Comment:
I seem to remember that other architectures export usart_send()/recv() with the same set of arguments, and then only output/input one byte